src/hotspot/share/jvmci/jvmciEnv.cpp

```diff
@@ -1622,7 +1622,9 @@ nmethod* JVMCIEnv::lookup_nmethod(address code, jlong compile_id_snapshot) {
   CodeBlob* cb = CodeCache::find_blob(code);
   if (cb == (CodeBlob*) code) {
     nmethod* nm = cb->as_nmethod_or_null();
-    if (nm != NULL && (compile_id_snapshot == 0 || nm->compile_id() == compile_id_snapshot)) {
+    assert(compile_id_snapshot >= 0, "negative compile id snapshot");
```

Comment:
That's something to check with the JVMCI experts but I'm wondering why `compile_id_snapshot` is a jlong whereas the compile id is a jint.

Reply:
It's a good question and unfortunately I cannot recall the reason. Maybe @tkrodriguez can. In any case, I don't think it impacts this change.

Reply:
That's far enough back that I don't really know why I used jlong.  Its type should really match the real compile id type whatever that is.  It's also fully internal to the JVMCI implementation so you can change it without affecting Graal.